applications/ShallowWaterApplication/custom_elements/conserved_var_element.cpp

```diff
@@ -0,0 +1,477 @@
+    { 
+        KRATOS_TRY 
+         
+        // Resize of the Left and Right Hand side 
+        unsigned int element_size = TNumNodes*3; 
+        if(rLeftHandSideMatrix.size1() != element_size) 
+            rLeftHandSideMatrix.resize(element_size,element_size,false); // Resizing the system in case it does not have the right size 
+         
+        if(rRightHandSideVector.size() != element_size) 
+            rRightHandSideVector.resize(element_size,false); 
+ 
+        // Getting gravity 
+        mGravity = rCurrentProcessInfo[GRAVITY_Z];
+        
+        // Getting properties
+        double manning2 = pow( GetProperties()[MANNING], 2 );
+        
+        // Getting water height unit converter
+        mHeightUnitConvert = rCurrentProcessInfo[WATER_HEIGHT_UNIT_CONVERTER];
+     
+        // Getting the time step (not fixed to allow variable time step) 
+        const double delta_t = rCurrentProcessInfo[DELTA_TIME];
+        const double dt_inv = 1.0 / delta_t; 
+        
+        // Compute the geometry
+        boost::numeric::ublas::bounded_matrix<double,TNumNodes, 2> DN_DX;
+        array_1d<double,TNumNodes> N;
+        double Area;
+        this-> CalculateGeometry(DN_DX,Area);
+        double elem_length = this->ComputeElemSize(DN_DX);
+        
+        // Getting the values of shape functions on Integration Points
+        boost::numeric::ublas::bounded_matrix<double,TNumNodes, TNumNodes> Ncontainer;  // In this case, number of Gauss points and number of nodes coincides
+        const GeometryType& rGeom = this->GetGeometry();
+        Ncontainer = rGeom.ShapeFunctionsValues( GeometryData::GI_GAUSS_2 );
+        
+        // Get nodal values for current step and projected variables (this function inlcudes the units conversion)
+        array_1d<double, TNumNodes*3> v_depth;
+        array_1d<double, TNumNodes*3> v_rain;
+        array_1d<double, TNumNodes*3> v_unknown;
+        array_1d<double, TNumNodes*3> v_proj_unknown;
+        GetNodalValues(v_depth, v_rain, v_unknown, v_proj_unknown);
+        
+        // Get element values (this function inlcudes the units conversion)
+        double height;
+        array_1d<double,2> height_grad;
+        array_1d<double,2> momentum;
+        double div_u;
+        GetElementValues(DN_DX, v_unknown, momentum, div_u, height, height_grad);
+        double abs_mom = norm_2(momentum);
+        double height73 = pow( height, 2.33333 );
+        if (height < 1e-2) height73 = 1e+4;
+        
+        // Compute stabilization and discontinuity capturing parameters
+        double tau_m;
+        double tau_h;
+        double k_dc;
+        ComputeStabilizationParameters(height, height_grad, elem_length, tau_m, tau_h, k_dc, rCurrentProcessInfo);
+        
+        // Some auxilary definitions
+        boost::numeric::ublas::bounded_matrix<double,2,TNumNodes*3> N_mom        = ZeroMatrix(2,TNumNodes*3);  // Shape functions matrix (for momentum unknown)
```

Comment:
Replace boost::numeric::ublas::bounded_matrix with bounded_matrix, we are going to do changes in the future..., and this do will help